pkg/sql/opt/memo/private_defs.go

```diff
@@ -31,9 +31,10 @@ type PrivateID uint32
 // operator. It provides the name and return type of the function, as well as a
 // pointer to an already resolved builtin overload definition.
 type FuncOpDef struct {
-	Name     string
-	Type     types.T
-	Overload *tree.Builtin
+	Name       string
+	Type       types.T
+	Properties *tree.FunctionProperties
```

Comment:
This feels clunky to me. I should be able to have a single pointer that holds onto a function definition, including its properties and its overloads. Why can't we have a `BuiltinDef` struct, like this:
```
type BuiltinDef struct {
  Props BuiltinProps
  Overloads []BuiltinOverload
}
```
This way, I can easily get a pointer to whatever I need - props, all overloads, any one overload, or the entire definition.